From: Roland Fehrenbacher <rf@q-leap.de>
Date: Tue, 28 Feb 2017 12:04:36 +0100
Subject: QL - Fixes in slurm_create_reservation

* Reservations didn't use a partition specified in reservation_dict
* Use uint32 for node_cnt and reservation flags
---
 pyslurm/pyslurm.pyx | 15 +++++++++------
 1 file changed, 9 insertions(+), 6 deletions(-)

diff --git a/pyslurm/pyslurm.pyx b/pyslurm/pyslurm.pyx
index a8a1720..f7d7ad2 100644
--- a/pyslurm/pyslurm.pyx
+++ b/pyslurm/pyslurm.pyx
@@ -3657,7 +3657,7 @@ def slurm_create_reservation(dict reservation_dict={}):
         int free_users = 0
         int free_accounts = 0
         unsigned int uint32_value
-        unsigned int time_value
+        slurm.time_t time_value
 
     slurm.slurm_init_resv_desc_msg(&resv_msg)
 
@@ -3668,9 +3668,9 @@ def slurm_create_reservation(dict reservation_dict={}):
     resv_msg.duration = uint32_value
 
     if reservation_dict[u'node_cnt'] != -1:
-        int_value = reservation_dict[u'node_cnt']
+        uint32_value = reservation_dict[u'node_cnt']
         resv_msg.node_cnt = <uint32_t*>slurm.xmalloc(sizeof(uint32_t) * 2)
-        resv_msg.node_cnt[0] = int_value
+        resv_msg.node_cnt[0] = uint32_value
         resv_msg.node_cnt[1] = 0
 
     if reservation_dict[u'node_list'] is not '':
@@ -3691,9 +3691,13 @@ def slurm_create_reservation(dict reservation_dict={}):
         name = reservation_dict[u'licenses']
         resv_msg.licenses = name
 
+    if reservation_dict[u'partition'] is not '':
+        name = reservation_dict[u'partition']
+        resv_msg.partition = name
+
     if reservation_dict[u'flags'] is not '':
-        int_value = reservation_dict[u'flags']
-        resv_msg.flags = int_value
+        uint32_value = reservation_dict[u'flags']
+        resv_msg.flags = uint32_value
 
     if reservation_dict[u'name'] is not '':
         name = reservation_dict[u'name']
@@ -3716,7 +3720,6 @@ def slurm_create_reservation(dict reservation_dict={}):
 
     return u"%s" % resID
 
-
 def slurm_update_reservation(dict reservation_dict={}):
     u"""Update a slurm reservation.
 
